context/app/static/js/components/workspaces/statusCodes.js

```diff
@@ -0,0 +1,49 @@
+// Workspaces:
+
+const workspaceStatus = {
+  active: {
+    isDone: false,
+  },
+  idle: {
+    isDone: false,
+  },
+  deleting: {
+    isDone: true,
+  },
+  error: {
+    isDone: true,
+  },
+};
+
+function validateWorkspaceStatus(status) {
+  const expected = Object.keys(workspaceStatus);
+  if (!expected.includes(status)) {
+    throw Error(`Unexpected workspace status "${status}"; expected one of (${expected})`);
+  }
+}
+
+// Jobs:
+
+const jobStatus = {
```

Comment:
```suggestion
const jobStatuses = {
```
Again, take or leave.